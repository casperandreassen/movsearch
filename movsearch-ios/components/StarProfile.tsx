import { useState } from 'react';
import { View, Text, StyleSheet, Image } from 'react-native';
import { Star } from '../GraphQL/MovieServices/types';
import { Appearance } from 'react-native';

type StarProfileProps = {
	star: Star;
};

/**
 * List item for stars in movie-details page.
 */

const StarProfile = ({ star }: StarProfileProps) => {
	const [imageSource, setImageSource] = useState(
		'https://image.tmdb.org/t/p/w200' + star.profile_path
	);

	return (
		<View style={styles.starContainer}>
			<Image
				key={String(star.name)}
				style={styles.starImage}
				source={{ uri: imageSource }}
				onError={() => {
					setImageSource(
						'https://www.kindpng.com/picc/m/643-6430001_no-profile-picture-girl-hd-png-download.png'
					);
				}}
			/>
			<Text
				style={[
					{
						fontSize: 13,
						fontWeight: '600',
						textAlign: 'center'
					},
					styles.darkMode
				]}
			>
				{star.name}
			</Text>
			<Text
				style={[
					{
						fontSize: 10,
						textAlign: 'center'
					},
					styles.darkMode
				]}
			>
				{star.character}
			</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	darkMode: {
		color: Appearance.getColorScheme() == 'dark' ? 'white' : 'black'
	},
	starContainer: {
		marginTop: 10,
		marginLeft: 4,
		marginBottom: 10,
		borderRadius: 4,
		borderColor: Appearance.getColorScheme() == 'dark' ? undefined : '#E0E0E0',
		borderWidth: 0.5,
		maxWidth: 130,
		backgroundColor: Appearance.getColorScheme() == 'dark' ? '#1B2430' : '#F9F9F9'
	},
	starImage: {
		width: 120,
		height: 180,
		borderTopLeftRadius: 4,
		borderTopRightRadius: 4,
		margin: 'auto'
	}
});

export default StarProfile;
