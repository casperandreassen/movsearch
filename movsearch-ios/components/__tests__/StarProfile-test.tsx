// __tests__

import React, { startTransition } from 'react';
import StarProfile from '../StarProfile';
import renderer from 'react-test-renderer';

const starMockData = {
	adult: false,
	gender: 2,
	id: 880,
	known_for_department: 'Acting',
	name: 'Ben Affleck',
	original_name: 'Ben Affleck',
	popularity: 30.916,
	profile_path: '/aTcqu8cI4wMohU17xTdqmXKTGrw.jpg',
	cast_id: 18,
	character: 'Bruce Wayne / Batman',
	credit_id: '52fe4d5bc3a368484e1e4c65',
	order: 0
};

const starInvalidMockData = {
	adult: false,
	gender: 2,
	id: 880,
	known_for_department: 'Acting',
	name: 'Ben Affleck',
	original_name: 'Ben Affleck',
	popularity: 30.916,
	profile_path: '/aTcqu8cI4wMohU17xTdXKTGrw.jpg',
	cast_id: 18,
	character: 'Bruce Wayne / Batman',
	credit_id: '52fe4d5bc3a368484e1e4c65',
	order: 0
};

it('renders correct star image', async () => {
	const tree = renderer.create(<StarProfile star={starMockData} />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renderes fallback image when star image path is broken', () => {
	const tree = renderer.create(<StarProfile star={starInvalidMockData} />);
	expect(tree).toMatchSnapshot();
});
