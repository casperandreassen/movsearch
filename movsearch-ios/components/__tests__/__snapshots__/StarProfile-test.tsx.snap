// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderes fallback image when star image path is broken 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F9F9F9",
      "borderColor": "#E0E0E0",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "marginBottom": 10,
      "marginLeft": 4,
      "marginTop": 10,
      "maxWidth": 130,
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w200/aTcqu8cI4wMohU17xTdXKTGrw.jpg",
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "height": 180,
        "margin": "auto",
        "width": 120,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 13,
          "fontWeight": "600",
          "textAlign": "center",
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    Ben Affleck
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 10,
          "textAlign": "center",
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    Bruce Wayne / Batman
  </Text>
</View>
`;

exports[`renders correct star image 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F9F9F9",
      "borderColor": "#E0E0E0",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "marginBottom": 10,
      "marginLeft": 4,
      "marginTop": 10,
      "maxWidth": 130,
    }
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w200/aTcqu8cI4wMohU17xTdqmXKTGrw.jpg",
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "height": 180,
        "margin": "auto",
        "width": 120,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontSize": 13,
          "fontWeight": "600",
          "textAlign": "center",
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    Ben Affleck
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 10,
          "textAlign": "center",
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    Bruce Wayne / Batman
  </Text>
</View>
`;
